examples/codes/xprotocol_with_goplugin_example/client_config.json

```diff
@@ -38,7 +38,7 @@
 										"upstream_protocol": "X",
 										"router_config_name": "client_router",
 										"extend_config": {
-											"sub_protocol": "x_example"
+											"sub_protocol": "codec"
```

Comment:
`x_example` is a better name for the sample.

better change the name here: https://github.com/mosn/mosn/blob/479b6bd2d03a22514e35b05588e8252ffd03d432/examples/codes/xprotocol_with_goplugin_example/codec/types.go#L26

Reply:
also `X` is no longer a protocol and `upstream_protocol` is useless now, better use this style:
```
"config": {
    "downstream_protocol": "x_example",
    "router_config_name": "client_router"
}
```

change details: https://mosn.io/blog/releases/v0.26.0/